extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -1297,12 +1294,61 @@ public static void consume(
                     .setValueToChunkObjectMapper(valueIngestData.toObjectChunkMapper);
         }
 
-        final KafkaStreamPublisher.Parameters publisherParameters = publisherParametersBuilder.build();
+        return new ConsumeStruct(tableDefinition, publisherParametersBuilder.build(), keyDeser, valueDeser);
+    }
+
+    /**
+     * Construct a {@link TableDefinition} based on the input Properties and {@link Consume.KeyOrValueSpec} parameters.
+     *
+     * @param kafkaProperties Properties to configure this table
+     * @param keySpec Conversion specification for Kafka record keys
+     * @param valueSpec Conversion specification for Kafka record values
+     * @return A TableDefinition derived from the input Properties and KeyOrValueSpec instances
+     */
+    @SuppressWarnings("unused")
+    public static TableDefinition getTableDefinition(
```

Comment:
We should explicitly call out the potential statefulness of `KeyOrValueSpec`. If the caller is going to rely on the correctness of this method, they _must_ call consume with the exact same instances of keySpec / valueSpec.

ie, the following is _incorrect_:

```java
TableDefinition tableDefinition = getTableDefinition(properties, null, avroSchema("MyValueSchema"));
Table table = consume(properties, ..., avroSchema("MyValueSchema"), ...)
```

it must be

```java
KeyOrValueSpec myValueSpec = avroSchema("MyValueSchema");
TableDefinition tableDefinition = getTableDefinition(properties, null, myValueSpec);
Table table = consume(properties, ..., myValueSpec, ...)
```

Reply:
Or gather, we should add documentation about the "guarantees" between getTableDefinition and consume; in what conditions is the table definition from getTableDefinition guaranteed to be the table definition of consume? (In my mind, it should be guaranteed when you use the exact same keySpec / valueSpec due to the statefulness of them.)